event-app-api/merge_intervals/merge_intervals.test.js

```diff
@@ -0,0 +1,200 @@
+import { mergeIntervals, findGaps } from "./merge_intervals.js";
+
+describe("Merge Intervals", () => {
+  test("Events with overlapping times returns one start and end time", () => {
+    const testEvents = [
+      {
+        starttime: 1679990400000,
+        endtime: 1680076800000,
+      },
+      {
+        starttime: 1680076800000,
+        endtime: 1680235200000,
+      },
+      {
+        starttime: 1680166400000,
+        endtime: 1683772800000,
+      },
+    ];
+    const expectedResult = [
+      {
+        starttime: 1679990400000,
+        endtime: 1683772800000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents)).toEqual(expectedResult);
+  });
+
+  test("Events with non-overlapping events returns both of the events", () => {
+    const testEvents2 = [
+      {
+        starttime: 1679930880000,
+        endtime: 1679952000000,
+      },
+      {
+        starttime: 1679952100000,
+        endtime: 1679973600000,
+      },
+    ];
+
+    const expectedResult = [
+      {
+        starttime: 1679930880000,
+        endtime: 1679952000000,
+      },
+      {
+        starttime: 1679952100000,
+        endtime: 1679973600000,
+      },
+    ];
+
+    expect(mergeIntervals(testEvents2)).toEqual(expectedResult);
+  });
+
+  test("Events with the same startime and endttime returns both of the events", () => {
```

Comment:
So this test case is when 2 events are 0 minutes in length because start and end times for each event individually is the same. However between the 2 of them they don't overlap, so it's just a subcase of the previous test.

You should add a test where start times for both events are the same and end times for both events are the same instead. Then the algorithm should only return 1 event.